packages/browser/src/integrations/useragent.ts

```diff
@@ -22,13 +22,14 @@ export class UserAgent implements Integration {
   public setupOnce(): void {
     addGlobalEventProcessor((event: Event) => {
       if (getCurrentHub().getIntegration(UserAgent)) {
-        if (!global.navigator || !global.location) {
+        if (!global.navigator || !global.location || !global.document) {
```

Comment:
guessing this is fine but not really sure - seems like it'd always be here if the others were here

Reply:
I believe that when the global is `self` (in a web worker), the first two exist but the last one doesn't: https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope.

What about something like this? That way we don't have to worry about which parts imply the existence of which other parts - as long as at least one is there, we add in whatever is defined.

```JavaScript
if (getCurrentHub().getIntegration(UserAgent)) {
  // if none of the information we want exists, don't bother
  if (!global.navigator && !global.location && !global.document) {
    return event;
  }

  // grab as much info as exists and add it to the event
  const url = event.request?.url || global.location?.href;

  const { referrer } = global.document || {};
  const { userAgent } = global.navigator || {};
  const headers = {
    ...event.request?.headers,
    ...(referrer && { Referer: referrer }),
    ...(userAgent && { 'User-Agent': userAgent }),
  };
  
  const request = { ...(url && { url }), headers };

  return { ...event, request };
}
```